gnd/src/main/java/com/google/android/gnd/repository/local/Edit.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.repository.local;
+
+import static java8.util.J8Arrays.stream;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.room.ColumnInfo;
+import androidx.room.Ignore;
+import androidx.room.TypeConverter;
+import com.google.android.gnd.vo.Record.Value;
+
+/** Fields describing operation type, old, and new values, common to Feature and Record edits. */
+public class Edit {
```

Comment:
`Edit` should be immutable (and probably implemented using AutoValue, since it is included as a property in other AutoValue classes so we would like to have generated `hashCode`, `equals`, etc. with similar value class semantics).

Reply:
I completely missed `Edit` when converting to @AutoValue. Good catch!